scenario 2.py

```diff
@@ -0,0 +1,91 @@
+import decimal
+import time
+from datetime import datetime
+import json
+from io import BytesIO, StringIO
+import logger
+
+logger = logger(__name__)
+
+def run(self):
+    
+    key = {"job_id": self.job_id}
+    result = False
+    start_time = datetime.now()
+    metadata = {
+        "source_file_type": self.source_file_type,
+        "pipeline_pattern": self.pipeline_pattern
+    }
+
+    """if file type is CSV."""
+    if self.source_file_type.lower() == "csv":
```

Comment:
use switch
JIT will only compile each case once making tit more performant for multiple checks than if/then/else if/else